Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -428,6 +445,196 @@ private Intent getLaunchIntent(Configuration configuration, String profileId, St
         return venmoIntent;
     }
 
+    /**
+     * Use this method with the manual browser switch integration pattern.
+     *
+     * @param browserSwitchResult a {@link BrowserSwitchResult} with a {@link BrowserSwitchStatus}
+     * @param callback            {@link VenmoOnActivityResultCallback}
+     */
+    public void onBrowserSwitchResult(@NonNull BrowserSwitchResult browserSwitchResult, @NonNull final VenmoOnActivityResultCallback callback) {
+        int result = browserSwitchResult.getStatus();
+        switch (result) {
+            case BrowserSwitchStatus.CANCELED:
+                braintreeClient.sendAnalyticsEvent("pay-with-venmo.browser-switch.canceled");
+                callback.onResult(null, new UserCanceledException("User canceled Venmo."));
+                break;
+            case BrowserSwitchStatus.SUCCESS:
+                Uri deepLinkUri = browserSwitchResult.getDeepLinkUrl();
+                if (deepLinkUri != null) {
+                    if (deepLinkUri.getPath().contains("success")) {
+                        braintreeClient.sendAnalyticsEvent("pay-with-venmo.browser-switch.success");
+                        String resourceIdFromBrowserSwitch = Uri.parse(String.valueOf(deepLinkUri)).getQueryParameter("resource_id");
+                        String paymentMethodNonceFromBrowserSwitch = Uri.parse(String.valueOf(deepLinkUri)).getQueryParameter("payment_method_nonce");
+                        String usernameFromBrowserSwitch = Uri.parse(String.valueOf(deepLinkUri)).getQueryParameter("username");
+
+                        String resourceIdFromAppSwitch = "";
+                        String paymentMethodNonceFromAppSwitch = "";
+                        String usernameFromAppSwitch = "";
+                        if (resourceIdFromBrowserSwitch == null && paymentMethodNonceFromBrowserSwitch == null && usernameFromBrowserSwitch == null) {
+                            String cleanedAppSwitchUri = String.valueOf(deepLinkUri).replaceFirst("&","?");
+                            resourceIdFromAppSwitch = Uri.parse(String.valueOf(cleanedAppSwitchUri)).getQueryParameter("resource_id");
+                            paymentMethodNonceFromAppSwitch = Uri.parse(String.valueOf(cleanedAppSwitchUri)).getQueryParameter("payment_method_nonce");
+                            usernameFromAppSwitch = Uri.parse(String.valueOf(cleanedAppSwitchUri)).getQueryParameter("username");
+                        }
```

Comment:
This logic doesn't _feel_ great but is due to getting 2 slightly different deeplink URLs back from the App vs Web based flow. If there is a cleaner way to do this I am down for sure!

App based `deepLinkUri`:
```
com.braintreepayments.demo.braintree://x-callback-url/vzero/auth/venmo/success&username=a-username&payment_method_nonce=fake-venmo-account-nonce&resource_id=a-resource-id
```

Web based `deepLinkUri`:
```
com.braintreepayments.demo.braintree://x-callback-url/vzero/auth/venmo/success?resource_id=a-resource-id&username=a-username&payment_method_nonce=fake-venmo-account-nonce
```

Specifically we cannot get the query parameters from the App based flow since the deep link uri we get back starts with an `&` instead of a `?`. Right now we are doing a replace on the first `&` if present and parsing the rest of the uri as we do on the web.

Reply:
Is that `&` instead of `?` intentional or something the Venmo team could fix? Otherwise this logic seems like the best option - but maybe for clarity we could break out all this URL parsing into helper methods in this class like `parseAppBasedReturnUrl` and `parseWebBasedReturnUrl`? Or something like that

Reply:
That's a good question if it's intentional or not. I can ask! Since we need all 3 things what kind of object would the helper methods return? Then we would need to parse the return types or duplicate the logic below. What do you think?

Reply:
Ah you're right there isn't a single return object, maybe is fine as is then 👍

Reply:
I ended up extracting the parsing logic into it's own methods (one for each value - nonce, username, resource id) here: https://github.com/braintree/braintree_android/pull/911/commits/b01835d10649518ab30edba7b588059970649444. I think it cleans this up a bit and hopefully makes it a bit more clear vs all living in this method. 🙏